### PR TITLE
Apply offset on MPI_Window 

### PR DESCRIPTION
**Pull request by janciesko**
This fixes errors for the MPI backend.

### CODE REVIEW

File: unit_tests/Test_PartitionedSubview.cpp [hunk 1]
```diff
@@ -230,7 +228,7 @@ void test_partitioned_subview2D_byRank(int i1, int i2) {
 TEST(TEST_CATEGORY, test_partitioned_subview) {
   // 1D subview
   test_partitioned_subview1D<int>(4, 4, 0, 0);
-  test_partitioned_subview1D<int>(50, 20, 8, 12);
+  test_partitioned_subview1D<int>(10, 10, 4, 4);
```
Comment (Rombur) [inline unit_tests/Test_PartitionedSubview.cpp:231]: Why did you change the test?

**Review by Rombur — APPROVED** [review]

File: unit_tests/Test_PartitionedSubview.cpp (hunk 1 above)
Comment (janciesko) [reply unit_tests/Test_PartitionedSubview.cpp]: Artifact from debugging. Restored the original test params.
